src/Specs.ts

```diff
@@ -97,6 +97,14 @@ export class Specs {
 			})
 		);
 
+		// HTTP
+		this._specs.appendSpec(
+			MODULES.SUBSTRATE,
+			this._specs.getSpec(CONFIG.HTTP_URL, 'Http URL', {
+				mandatory: false,
```

Comment:
Nit: Should this also have a `regexp` to properly validate the HTTP URL? Similar to `WS` to ensure we can't pass arbitrary URLs without port numbers?

Reply:
Yea I agree, I had it before but it would fail everytime because if there was no default addr it would see an empty value and fail. But given the suggestions below I think it will be much easier :)